src/main/java/org/veupathdb/service/eda/compute/plugins/correlationassaymetadata/CorrelationAssayMetadataPlugin.java

```diff
@@ -0,0 +1,151 @@
+import org.veupathdb.service.eda.common.model.EntityDef;
+import org.veupathdb.service.eda.common.model.ReferenceMetadata;
+import org.veupathdb.service.eda.common.model.VariableDef;
+import org.veupathdb.service.eda.common.plugin.util.PluginUtil;
+import org.veupathdb.service.eda.compute.RServe;
+import org.veupathdb.service.eda.compute.plugins.AbstractPlugin;
+import org.veupathdb.service.eda.compute.plugins.PluginContext;
+import org.veupathdb.service.eda.generated.model.CorrelationComputeConfig;
+import org.veupathdb.service.eda.generated.model.CorrelationPluginRequest;
+import org.veupathdb.service.eda.generated.model.VariableSpec;
+import org.veupathdb.service.eda.generated.model.APIVariableDataShape;
+import org.veupathdb.service.eda.generated.model.CollectionSpec;
+
+import java.io.InputStream;
+import java.util.stream.Stream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import static org.veupathdb.service.eda.common.plugin.util.PluginUtil.singleQuote;
+
+public class CorrelationAssayMetadataPlugin extends AbstractPlugin<CorrelationPluginRequest, CorrelationComputeConfig> {
+
+  private static final String INPUT_DATA = "correlation_input";
+
+  public CorrelationAssayMetadataPlugin(@NotNull PluginContext<CorrelationPluginRequest, CorrelationComputeConfig> context) {
+    super(context);
+  }
+
+  @NotNull
+  @Override
+  public List<StreamSpec> getStreamSpecs() {
+    // Get the collection variable and its entity
+    CorrelationComputeConfig computeConfig = getConfig();
+    CollectionSpec collectionVariable1 = computeConfig.getCollectionVariable1();
+    String entityId = collectionVariable1.getEntityId();
+
+    // Wrangle into correct types for what follows
+    EntityDef entity = getContext().getReferenceMetadata().getEntity(entityId).orElseThrow();
+    VariableSpec collectionVariable1VarSpec = VariableDef.newVariableSpec(entityId, collectionVariable1.getCollectionId());
+
+
+    // Grab all continuous variabls from ancestors
+    // The next line only grabs the second-up ancestor instead of all of them, because when i grabbed them all
+    // the variables wold be duplicated in the output. I tried distinct() but that didn't work. Any ideas?
+    // I also noticed we were receiving stable ids in the oriringal request, so i've filtered them out below. There 
+    // is probably a nicer way to do this.
+    // This at least works for now!
+    Stream<VariableDef> descendantVariableStream = getContext().getReferenceMetadata().getAncestors(entity).get(1).getVariables().stream();
+    List<VariableDef> metadataVariables = descendantVariableStream.filter(var -> var.getDataShape() == APIVariableDataShape.CONTINUOUS).filter(var -> !var.getVariableId().contains("_stable_id")).toList();
+
+
+    return List.of(new StreamSpec(INPUT_DATA, getConfig().getCollectionVariable1().getEntityId())
+        .addVars(getUtil().getChildrenVariables(collectionVariable1VarSpec))
+        .addVars(metadataVariables)
```

Comment:
im wondering if we should be using two streamSpecs for these plugins? maybe weve thought about it before, had reasons not to, and ive forgotten?

Reply:
discussed in slack, decided we wont worry about it yet.